KeyVaultCA.Web/Startup.cs

```diff
@@ -29,23 +29,27 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            var caConfig = new CAConfiguration();
-
-            var keyVaultServiceClient = new KeyVaultServiceClient(caConfig.KeyVaultUrl);
-            keyVaultServiceClient.SetAuthenticationClientCredential(caConfig.AppId, caConfig.Secret);
-            var kvCertProvider = new KeyVaultCertificateProvider(keyVaultServiceClient);
-
+            services.AddApplicationInsightsTelemetry();
+
+            var estConfig = Configuration.GetSection("KeyVault").Get<EstConfiguration>();

```

Comment:
Just to confirm: will this also work when passing the configuration through environment variables or would appsettings.json be the only way?

Reply:
It will work (in the order of precedence) with appsettings.json, env variables and command line args.

Reply:
ok, thanks. so the environment variable value needs to be passed in as json string, right?

Reply:
Yes, you could put it in the `launchsettings.json` or could even set is at system level, if desired. I think an easy approach would be to load all the common settings (like the ones for KV access) from the `appsettings.json` and override specific ones (like `IsRootCA`) from command line. Would that be ok?

Reply:
thanks yes. I was just thinking about the configuration in App Service for example. The Key/Value pairs are exposed to the app as ENVs. So the value would need to be a json object, so that `GetSection("KeyVault").Get<EstConfiguration>()` works or one deploys the appsettings.json file although it would not be possible to use [KeyVault references](https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references) then.

Reply:
If specified, the settings from the App Service take precedence over the local ones and they should be in a similar format "Section" separator "SettingName" with separator being `:` or `__` between them depending on OS.

Reply:
If the name convention is respected, the value of the settings can also be retrieved from the KV.

Reply:
got it, thanks!